src/test/java/com/devonfw/sample/archunit/LayerDependencyRulesTest.java

```diff
@@ -0,0 +1,97 @@
+package com.devonfw.sample.archunit;
+
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
+
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class LayerDependencyRulesTest {
+    
+    // 'access' catches only violations by real accesses, i.e. accessing a field, calling a method; compare 'dependOn' further down
+
+    // L09: verifying that code from logic layer does not access service layer (of same app).
+    @ArchTest
+    static final ArchRule logic_should_not_access_services =
+            noClasses().that().resideInAPackage("..logic..")
+                    .should().accessClassesThat().resideInAPackage("..service..");
+
+    // L10: verifying that dataaccess layer does not access service layer.
+    // L12: verifying that dataaccess layer does not access logic layer.
+    @ArchTest
+    static final ArchRule persistence_should_not_access_services_or_logic =
+            noClasses().that().resideInAPackage("..dataaccess..")
+                    .should().accessClassesThat().resideInAnyPackage("..service..", "..logic..");
+
+    @ArchTest
+    static final ArchRule services_should_only_be_accessed_by_clients_or_other_services =
+            classes().that().resideInAPackage("..service..")
+                    .should().onlyBeAccessed().byAnyPackage("..client..", "..service..");
+
+    @ArchTest
+    static final ArchRule services_should_only_access_logic_common_or_other_services =
+            classes().that().resideInAPackage("..service..")
+                    .should().onlyAccessClassesThat().resideInAnyPackage("..service..", "..common..", "..logic..", "java..", "javax..");
```

Comment:
If we start whitelisting what packages are allowed to reference (like `java.*` and `javax.*`) we will end up in a maintenance hell. Thus the `mayOnlyBeAccessedByLayers` will make it much easier for us to only say what layer can access what other layers.

Reply:
In reference to your comment from above and this one:

Thank you for the hint.
This helped me to figure out why my first approach led to plenty violations of dependencies with reference to java.*, javax.*, org.mapstruct.* etc.

I've updated my code with respect to your suggestions and above violations have been resolved.
 - I deleted LayerDependencyRulesTest.java 
 - and adjusted ArchitectureTest.java to use correct layer definitions and utilizing mayOnlyBeAccessedByLayers rules.

Upon above changes I also updated my PR #14 . All 12 intended violations are detected.

Reply:
Thanks, done and resolved.